app/helpers/email_signup_helper.rb

```diff
@@ -0,0 +1,24 @@
+module EmailSignupHelper
+
+  def email_signup_path(atom_url)
+    special_case = special_cases.find { |c| atom_url.match(c[:regex]) }
+
+    if special_case
+      special_case[:email_signup_path]
+    else
+      new_email_signups_path(email_signup: {feed: atom_url})
+    end
+  end
+
+private
+
+  def special_cases
+    [
+      {
+        regex: /medicines-and-healthcare-products-regulatory-agency.atom$/,
+        email_signup_path: "/government/organisations/medicines-and-healthcare-products-regulatory-agency/email-signup"
```

Comment:
Better to use a routing helpers instead of hardcoding the path, e.g. `organisation_email_signup_information('medicines-and-healthcare-products-regulatory-agency')`.